frontend-react/src/pages/submissions/SubmissionDetails.test.tsx

```diff
@@ -0,0 +1,115 @@
+import { MatcherFunction, screen } from "@testing-library/react";
+import ActionDetailsResource from "../../resources/ActionDetailsResource";
+import { success } from "../../resources/fixtures/ActionDetailFixtures";
+import { renderWithRouter } from "../../utils/CustomRenderUtils";
+
+import SubmissionDetails, {
+    DestinationItem,
+    DetailItem,
+} from "./SubmissionDetails";
+
+/*
+    Using the included regex can end up pulling various elements where the
+    value has the parsed timestamp. Use a function
+*/
+const timeRegex: RegExp = /[0-9]{1,2}:[0-9]{1,2} [A,P]M/;
+const mockData: ActionDetailsResource = ActionDetailsResource.dummy();
+jest.mock("rest-hooks", () => ({
+    useResource: () => {
+        return mockData;
+    },
+    /* Must return children when mocking, otherwise nothing inside renders */
+    NetworkErrorBoundary: ({ children }: { children: JSX.Element[] }) => {
+        return <>{children}</>;
+    },
+}));
+
+describe("SubmissionDetails", () => {
+    beforeEach(() => {
+        /*
+            TODO: Custom renderers to handle Router, etc. when
+            we need to use more complex render settings
+        */
+        renderWithRouter(<SubmissionDetails />);
+    });
+
+    test("renders without error", async () => {
+        const container = await screen.findByTestId("container");
+        expect(container).toBeInTheDocument();
+    });
+
+    test("renders data to sub-components", async () => {
+        /* Custom matcher for transitionTime */
+        const findTimeWithoutDate: MatcherFunction = (
+            content,
+            element
+        ): boolean => {
+            if (!content.includes("7 Apr 1970") && timeRegex.test(content))
+                return true;
+            return false;
+        };
+
+        /* Report ID DetailItem */
+        const idElement = await screen.findByText(mockData.id);
+
+        /* DestinationItem contents*/
+        const receiverOrgName = await screen.findByText(
+            mockData.destinations[0].organization
+        );
+        const transmissionDate = await screen.findByText("7 Apr 1970");
+        const transmissionTime = screen.getByText(findTimeWithoutDate);
+        const recordsTransmitted = await screen.findByText(
+            mockData.destinations[0].itemCount
+        );
+
+        /*
+            As above, so below. Add any new elements needing unit test
+            verification to this array to be tested!
+        */
+        const testElements = [
+            idElement,
+            receiverOrgName,
+            transmissionDate,
+            transmissionTime,
+            recordsTransmitted,
+        ];
+
+        for (let i = 0; i < testElements.length; i++) {
```

Comment:
can we use `for of` here?

Reply:
For..of worked! Thanks :) I was using for..in 🤦🏼